File: Dal/Models/Slot.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using BPGezinswetenschappen.DAL.Models;

namespace DAL.Models
{ 
    public class Slot
    {
        public int SlotId { get; set; }
        public TimeOnly StartTime { get; set; }
        public TimeOnly EndTime { get; set; }

        public int? PresentationDayId { get; set; }
        public PresentationDay? PresentationDay { get; set; }

        public int? ClassroomId { get; set; }
        public Classroom? Classroom { get; set; }

        public ICollection<PresentationSlot> Presentations { get; set; } = new List<PresentationSlot>();
        public ICollection<UserSlot> Availabilities { get; set; } = new List<UserSlot>();

    }


}

File: Client/src/app/shared/models/slot.ts
import { PresentationDay } from './presentationday';
import { Classroom } from './classroom';
import { Presentation } from './presentation';

// Note: To avoid circular reference issues, use 'any' for related types for now.
export interface Slot {
  slotId: number;
  startTime: string; // Use string for TimeOnly (e.g. 'HH:mm:ss')
  endTime: string; // Use string for TimeOnly
  presentationDayId?: number;
  presentationDay?: any;
  classroomId?: number;
  classroom?: Classroom;
  presentations: Presentation[];
  availabilities: any[];
}
